### ISSUE TITLE
[bug]: Storybook navigation has focus issues

### ISSUE THREAD

**Issue by leagrdv** [swisspost/design-system#5256]
### 🐞 Describe the bug

The storybook navigation is not accessible as some accessibility requirements are not met:
- The focus is not visible on any of the navigation items
- When opening a section of the navigation, we cannot navigate to its children
- Same goes for the settings: when opening the Settings dropdown (⚙️ at the top of the navigation), the elements within are not accessible using the keyboard)

**Comment by leagrdv** [comment]
Follow-up II on that ticket:
- PR is now only fixing the first issue
- I've opened a ticket for the [second issue on storybook's github](https://github.com/storybookjs/storybook/issues/31267)
